Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/network/NetworkStateProvider.kt

```diff
@@ -90,34 +92,46 @@ class NetworkStateProvider @Inject constructor(
         )
 
     private val currentState: State
-        get() = manager.activeNetwork.let { network ->
+        @SuppressLint("NewApi")
+        get() = if (BuildVersionWrap.hasAPILevel(Build.VERSION_CODES.M)) {
+            manager.activeNetwork.let { network ->
+                State(
+                    activeNetwork = network,
+                    capabilities = network?.let {
+                        try {
+                            manager.getNetworkCapabilities(it)
+                        } catch (e: SecurityException) {
+                            Timber.tag(TAG).e(e, "Failed to determine network capabilities.")
+                            null
+                        }
+                    },
+                    linkProperties = network?.let {
+                        try {
+                            manager.getLinkProperties(it)
+                        } catch (e: Exception) {
+                            Timber.tag(TAG).e(e, "Failed to determine link properties.")
+                            null
+                        }
+                    },
+                    assumeMeteredConnection = testSettings.fakeMeteredConnection.value
+                )
+            }
+        } else {
+            // Most state information is not available
             State(
-                activeNetwork = network,
-                capabilities = network?.let {
-                    try {
-                        manager.getNetworkCapabilities(it)
-                    } catch (e: SecurityException) {
-                        Timber.tag(TAG).e(e, "Failed to determine network capabilities.")
-                        null
-                    }
-                },
-                linkProperties = network?.let {
-                    try {
-                        manager.getLinkProperties(it)
-                    } catch (e: Exception) {
-                        Timber.tag(TAG).e(e, "Failed to determine link properties.")
-                        null
-                    }
-                },
-                isFakeMeteredConnection = testSettings.fakeMeteredConnection.value
+                activeNetwork = null,
+                capabilities = null,
+                linkProperties = null,
```

Comment:
are there any alternatives on pre API 23 for these nullables ?

Reply:
We could additionally store the active `NetworkInfo` object (deprecated), but this would be a bit confusing; and since the fields currently doesn't seem to be needed anywhere else, it might be better to find out how to extract one specific thing that is needed once it is needed.

Reply:
Sounds good 👍